dali/operators/image/convolution/filter.cc

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+
+#include "dali/operators/image/convolution/filter.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+DALI_SCHEMA(experimental__Filter)
+    .DocStr(R"code(Convolves the image with the provided filter.
+
+The operator requires two positional arguments: the batch of samples and the batch of filters.
+
+Sample can be an image, a video or volumetric (3D) data.
+Samples can contain channels: channel-first and channel-last layouts are supported.
+In case of video/sequences, the frame extent must preced the channels extent, i.e.,
+for example, a video with ``"FCHW"`` layout is supported, but ``"CFHW"`` samples are not.
+
+Samples with the following types are supported:
+int8, int16, uint8, uint16, float16, float32.
+Please note that the intermediate type used for the computation is always float32.
+
+For inputs with two spatial dimensions (images or video), the filters must be 2D arrays
+(or a sequence of 2D arrays to be applied
+:func:`per-frame<nvidia.dali.fn.per_frame>` to a video input).
+For volumetric inputs, the filter must be a 3D array.
+The filter values must have float32 type.
+
+If the optional third positional argument is specified, it must be a batch of scalars.
+If ``"border"`` is set to ``"constant"``, the input samples will be padded with
+the corresponding scalars when convolved with the filter.
+The scalars must be of the same type as the input samples.
+For video/sequence input, an array of scalars can be specified to be applied
+:func:`per-frame<nvidia.dali.fn.per_frame>`.
+
+.. note::
+  In fact, the operator computes a correlation, not a convolution,
+  i.e. the order of filter elements is not flipped when computing product of
+  the filter and the image.
+
+)code")
+    .NumInput(2, 3)
```

Comment:
```suggestion
    .NumInput(2, 3)
    .InputDox(0, "data", "TensorList", R"code(Batch of input data.)code")
    .InputDox(1, "filter", "TensorList", R"code(Batch of filters.)code")
    .InputDox(2, ...)    
```
Please add documentation of the inputs, and perhaps remove the sentence in line 25.

Reply:
Done, I moved whole sections from the main description to relevant parameters' docs.